### ISSUE TITLE
Recommended way to perform inference?

### ISSUE THREAD

**Issue by joel1st** [zhengqili/UprightNet#2]
First off thanks for putting an implementation of the paper on Github! 

Is there a recommended way of performing inference on a non-dataset (Interiornet/Scannet) image? I have attempted creating a new Dataloader that is basically the same as the Interiornet loader (but with all labelling logic removed, as no training is required). 

I've also added a new method to the uprightnet model: 
```
    def infer_model(self, input_, targets):
        # switch to evaluation mode
        with torch.no_grad():
            input_imgs = Variable(input_.cuda() , requires_grad = False)

            pred_cam_geo, pred_up_geo, pred_weights = self.netG.forward(input_imgs)
            # normalize predicted surface nomral
            pred_cam_geo_unit = self.criterion_joint.normalize_coords(pred_cam_geo)
            pred_up_geo_unit = self.criterion_joint.normalize_normal(pred_up_geo)

        return pred_cam_geo_unit, pred_up_geo_unit
```

The script runs without any error, but the pred_cam_geo_unit, and pred_up_geo_unit is not what I was expecting. I would have assumed this would give me the camera pose (or camera direction vector), instead I get back a matrix with two of the dimensions the same shape as the resized image: torch.Size([1, 9, 288, 384]) torch.Size([1, 3, 288, 384]). Am I interpretting this incorrectly or is there a different way that I need to perform inference?

It might also be worth noting that the way I performed inference, is I downloaded the 'checkpoints.zip' and ran `python test.py --mode ResNet --dataset interiornet`. But in test.py I changed it to load my image dataloader instead of interiornet (and then called the function above rather than the test function).

**Comment by zhengqili** [comment]
Hi,

Yes. the network provide you with surface frame predictions in both local and upright coordinate system rather than final camera pose. In order to get final upright directional vector, go to function "compute_angle_error" in networks.py. In this function, you will see the variable "est_up_n", which is the estimated up vector computed by constrained weighted least square as described in the paper.

**Comment by joel1st** [comment]
Hello @zhengqili - Thanks for that. That works perfectly. Now I just need to try estimate the height (or position) of the camera relative to the ground. My thought is to solve for the height using the gradient of the floor/ground + the camera orientation. Not exactly sure how to go about doing this yet..

**Comment by oleksandrmilko** [comment]
@joel1st 
Good day! Can you share the code for performing the inference you ended up with, please? Many thanks
